analyses/cell-type-ewings/exploratory_analysis/annotation_notebooks/SCPCS000492/SCPCL000824_tumor-gene-expression-exploration.Rmd

```diff
@@ -0,0 +1,749 @@
+It looks like we get pretty good agreement between our original classification and the updated `AUCell` classification using the marker gene list.
+
+Now we repeat on `SCPCL000824`. 
+
+```{r}
+# get new auc and plot thresholds 
+new_auc <- AUCell::AUCell_run(counts(sce), collection)
+new_assignment <- AUCell::AUCell_exploreThresholds(new_auc, assign = TRUE)
+```
+
+Here we do not see any bimodal distributions and that most cells are considered to not be expressing the specified gene sets, which doesn't fit with our previous findings. 
+I expect that without a bimodal distribution this model does not do a good job of calculating a threshold. 
+
+```{r}
+ref_auc_df <- ref_auc@assays@data$AUC |> 
+  as.data.frame() |> 
+  tibble::rownames_to_column("gene_set") |> 
+  tidyr::pivot_longer(!gene_set, 
+                      names_to = "barcodes", 
+                      values_to = "auc") |> 
+  #dplyr::filter(gene_set == "markers") |> 
+  dplyr::mutate(sample = "SCPCL000822")
+
+new_auc_df <- new_auc@assays@data$AUC |> 
+  as.data.frame() |> 
+  tibble::rownames_to_column("gene_set") |> 
+  tidyr::pivot_longer(!gene_set, 
+                      names_to = "barcodes", 
+                      values_to = "auc") |> 
+  #dplyr::filter(gene_set == "markers") |> 
+  dplyr::mutate(sample = "SCPCL000824")
+
+all_auc_df <- dplyr::bind_rows(list(ref_auc_df, new_auc_df))
+
+ggplot(all_auc_df, aes(x = auc, color = sample)) +
+  geom_density() +
+  facet_grid(rows = vars(gene_set))
+```
+
+Looking at the AUC across both samples for all gene sets, we see that they are in the same range of values and tend to mirror each other. 
+The main difference here is the lack of bimodal distribution in `SCPCL000824`. 
+
+Let's use the threshold identified by `AUCell` for `SCPCL000822` to identify tumor cells in `SCPCL000824`.  
+Here, we label any cells that have passed the threshold set for the marker gene sets as tumor cells. 
+
+```{r}
+# pull out threshold used for assigning cells from ref sample 
+ref_threshold <- ref_assignment$markers$aucThr$selected
+
+# create a vector of tumor cells in 824 using the cutoff from 822
+new_tumor_cells <- new_auc_df |> 
+  dplyr::filter(gene_set == "markers", auc >= ref_threshold) |>
+  dplyr::pull(barcodes)
+
+# add to original classification df 
+classification_df <- classification_df |> 
+  dplyr::mutate(auc_classification = dplyr::if_else(barcodes %in% new_tumor_cells, "Tumor", "Normal"))
+```
+
+```{r}
+# visualize which cells are classified as tumor 
+ggplot(classification_df, aes(x = UMAP1, y = UMAP2, color = auc_classification)) +
+  geom_point(alpha = 0.5, size = 0.5)
+```
+
+This looks similar to using the updated marker gene cutoff to classify cells where most cells in that bottom group are tumor cells.  
+As a reminder, the updated marker gene classification is the classification of tumor cells based on the marker gene expression cutoff determined in `SCPCL000822`. 
+We can confirm this by calculating the confusion matrix. 
```

Comment:
Just noting that in the confusion matrix it is hard to tell which is which... You might rename one of them for clarity?

The reason I was interested was that I kind of wanted to know which one was more "conservative" as there does seem to be a pretty substantial difference. I'm not sure what I would do with that info though.

> Update: This is more visible in the later plots: AUCell seems to be the one that is a bit more conservative in this case, calling more cells as normal.

Reply:
> Just noting that in the confusion matrix it is hard to tell which is which... You might rename one of them for clarity?

I don't believe we can do this since the classes are supposed to have the same names to be able to compare. I get an error about classes not being in the same order if I change any of the names of the classes. But I can add in clarification in the text.